src/main.rs

```diff
@@ -47,9 +53,30 @@ fn main() {
     let opts = cli::RunOptions::from_args();
     info!("opts: {:#?}", opts);
 
-    info!("reading {:?}", opts.module);
-    // TODO: don't just panic here...
-    let mut reader = File::open(&opts.module).expect("Unable to open file");
+    cfg_if! {
+        if #[cfg(unix)] {
+            let mut reader = match opts.module_on_fd {
+                Some(fd) => {
+                    info!("reading module from fd {:?}", fd);
+                    // SAFETY: unsafe if something is using the given fd.
+                    // parse_module_fd() enforces fd >= 3, and nothing above
+                    // opens/duplicates new file descriptors, so we're OK.
+                    unsafe { File::from_raw_fd(fd) }
+                },
+                None => {
+                    let path = opts.module.expect("required_unless failure");
+                    info!("reading module from {:?}", path);
+                    File::open(&path)
+                        .with_context(|| format!("failed opening {:?}", path))?
+                },
+            };
+        } else {
+            let path = opts.module.expect("missing required arg");
+            info!("reading module from {:?}", path);
+            let mut reader = File::open(&path)
+                .with_context(|| format!("failed opening {:?}", path))?;
+        }
+    }
```

Comment:
I don't like `cfg_if!` in general. And it doesn't make sense here either. Let's find a way to split this more reasonably and drop the dependency.

Reply:
Pushed an updated version that drops `cfg_if` in favor of a helper function.